modules/simple_forms_api/app/services/simple_forms_api/form_remediation/submission_remediation_data.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'simple_forms_api/form_remediation/configuration/base'
+
+module SimpleFormsApi
+  module FormRemediation
+    class SubmissionRemediationData
+      attr_reader :file_path, :submission, :attachments, :metadata
+
+      def initialize(id:, config: Configuration::Base.new)
+        @config = config
+
+        validate_input(id)
+        fetch_submission(id)
+
+        @attachments = []
+        @metadata = {}
+      rescue => e
+        config.handle_error("#{self.class.name} initialization failed", e)
+      end
+
+      def hydrate!
+        form_number = fetch_submission_form_number
+        form = build_form(form_number)
+        filler = PdfFiller.new(form_number:, form:)
+
+        handle_submission_data(filler, form, form_number)
+        self
+      rescue => e
+        config.handle_error('Error hydrating submission', e)
+      end
+
+      private
+
+      attr_reader :config
+
+      def validate_input(id)
+        raise ArgumentError, "No #{config.id_type} was provided" unless id
+      end
+
+      def fetch_submission(id)
+        @submission = config.submission_type.find_by(config.id_type => id)
+        validate_submission
+      end
+
+      def validate_submission
+        raise 'Submission was not found or invalid' unless submission&.send(config.id_type)
+        raise "#{self.class} cannot be built: Only VFF forms are supported" unless vff_form?
+      end
+
+      def fetch_submission_form_number
+        vff_forms_map.fetch(submission.form_type)
+      end
+
+      def build_form(form_number)
+        form_class_name = "SimpleFormsApi::#{form_number.titleize.delete(' ')}"
+        form_class = form_class_name.constantize
+        form_class.new(form_data_hash)
+      rescue NameError => e
+        config.handle_error("Form class not found for #{form_class_name}", e)
+      end
+
+      def handle_submission_data(filler, form, form_number)
+        @file_path = generate_pdf_file(filler)
+        @metadata = validate_metadata(form)
+        @attachments = process_attachments(form, form_number)
+      end
+
+      def generate_pdf_file(filler)
+        filler.generate(timestamp: submission.created_at)
+      rescue => e
+        config.handle_error('Error generating filled submission PDF', e)
+      end
+
+      def validate_metadata(form)
+        SimpleFormsApiSubmission::MetadataValidator.validate(
+          form.metadata,
+          zip_code_is_us_based: form.zip_code_is_us_based
+        )
+      rescue => e
+        config.handle_error('Metadata validation failed', e)
+      end
+
+      def process_attachments(form, form_number)
```

Comment:
optional comment - just thinking - wondering if its possible for each consumer to specify this, so that we don't have to manage it here.

Reply:
Do you mean allow them to override this method and its functionality or give them an option to skip this step?

Reply:
more on the lines of, I'm thinking of... is there a way to write this in a way so that we never have to touch this function again. vs updating this for every new form with uploads.

Reply:
Ahh, yes. There is but it will require refactoring of external form classes. I already have a WIP draft PR out there to address this. I'll eventually get back to it 😂 .